internal/kafka/internal/services/kafka.go

```diff
@@ -944,7 +945,13 @@ func (k *kafkaService) ChangeKafkaCNAMErecords(kafkaRequest *dbapi.KafkaRequest,
 		AccessKeyID:     k.awsConfig.Route53AccessKey,
 		SecretAccessKey: k.awsConfig.Route53SecretAccessKey,
 	}
-	awsClient, err := k.awsClientFactory.NewClient(awsConfig, kafkaRequest.Region)
+
+	route53Region, err := k.getRoute53RegionFromKafkaRequest(kafkaRequest)
+	if err != nil {
+		return nil, errors.NewWithCause(errors.ErrorGeneral, err, "error getting route 53 region from kafka request")
+	}
+
+	awsClient, err := k.awsClientFactory.NewClient(awsConfig, route53Region)
```

Comment:
I was thinking that it might be a good idea to change the type of the aws client and instead of being `type Client` rename it to `type Route53Client` as the methods in there are only related to Route53. If in the future we need to interact with another AWS API for example EC2 then we would create another interface `type EC2Client` and so on. That would separate the interfaces by AWS service, similarly to what AWS Go SDK does

Reply:
I think it is useful, we will likely want to offer Google Cloud DNS at some point in the future, at least for GCP clusters.